lib/Geocode.php

```diff
@@ -1612,7 +1618,7 @@ function lookup()
 					if ($this->bIncludePolygonAsKML) $sSQL .= ",ST_AsKML(geometry) as askml";
 					if ($this->bIncludePolygonAsSVG) $sSQL .= ",ST_AsSVG(geometry) as assvg";
 					if ($this->bIncludePolygonAsText || $this->bIncludePolygonAsPoints) $sSQL .= ",ST_AsText(geometry) as astext";
-					$sSQL .= " from placex where place_id = ".$aResult['place_id'];
+					$sSQL .= " from (select place_id,centroid,ST_SimplifyPreserveTopology(geometry,".$this->fPolygonSimplificationThreshold.") as geometry from placex where place_id = ".$aResult['place_id'].") as plx";
```

Comment:
On Tue, Apr 28, 2015 at 10:19 PM, Sarah Hoffmann <notifications@github.com>
wrote:

> In lib/Geocode.php
> <https://github.com/twain47/Nominatim/pull/267#discussion_r29283700>:
>
> > @@ -1612,7 +1618,7 @@ function lookup()
> >  					if ($this->bIncludePolygonAsKML) $sSQL .= ",ST_AsKML(geometry) as askml";
> >  					if ($this->bIncludePolygonAsSVG) $sSQL .= ",ST_AsSVG(geometry) as assvg";
> >  					if ($this->bIncludePolygonAsText || $this->bIncludePolygonAsPoints) $sSQL .= ",ST_AsText(geometry) as astext";
> > -					$sSQL .= " from placex where place_id = ".$aResult['place_id'];
> > +					$sSQL .= " from (select place_id,centroid,ST_SimplifyPreserveTopology(geometry,".$this->fPolygonSimplificationThreshold.") as geometry from placex where place_id = ".$aResult['place_id'].") as plx";
>
> Problem is that SimplifyPreserveTopology is not cheap. For Germany I get
> 25 ms on average for SELECT geometry FROM placex WHERE place_id = x and
> 1073 ms for select geometry from (select
> ST_SimplifyPreserveTopology(geometry, 0.0) as geometry from placex where
> place_id = x) as p. I would check for x > 0.0.
>
I've had some different figures, but I agree if the simplify function is
non-trivial in case of 0.0, we should check that ourselves.  Should I
submit an update to this pull request or a new one?